projects/aws/upgrader/upgrade/syscalls.go

```diff
@@ -0,0 +1,56 @@
+//go:generate ../hack/tools/bin/mockgen -destination ./mocks/syscalls_mock.go -package mocks . SysCalls
+
+package upgrade
+
+import (
+	"context"
+	"io/fs"
+	"os"
+	"os/exec"
+)
+
+type SysCalls interface {
+	WriteFile(string, []byte, os.FileMode) error
+	ReadFile(string) ([]byte, error)
+	OpenFile(string, int, os.FileMode) (*os.File, error)
+	Stat(string) (os.FileInfo, error)
+	Executable() (string, error)
+	ExecCommand(context.Context, string, ...string) ([]byte, error)
+	MkdirAll(string, os.FileMode) error
+}
+
+type sysCalls struct {
+	SysCalls
```

Comment:
This doesnt seem right. The empty struct already implements the interface, why embed the interface? Also, you dont need an interface at all.

Reply:
The idea was to use an interface constructor and i need it for the testability reasons stated above. But i don't think i need to embed it though. I will clean up!